pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -616,3 +631,17 @@ func targetMatch(target *configv1.Release, to string, toImage string) (bool, err
 
 	return false, nil
 }
+
+func patchDesiredUpdate(update *configv1.Update, client configv1client.Interface, clusterVersionName string) error {
```

Comment:
nit, to reduce the effort of eventually getting `Context` wired up for this `Patch` call, I'd prefer opening with a `ctx context.Context` argument here, and pushing the `TODO()` initialization as far up the stack as we conveniently can (but no need to restructure call signatures on existing functions).